src/datachain/lib/listing.py

```diff
@@ -97,15 +109,68 @@ async def _scandir(client, prefix, fetch_workers) -> AsyncIterator:
     await main_task
 
 
-def list_bucket(uri: str, client_config=None, fetch_workers=FETCH_WORKERS) -> Callable:
+def list_bucket(uri: str, fetch_workers=FETCH_WORKERS, **kwargs) -> Callable:
     """
     Function that returns another generator function that yields File objects
     from bucket where each File represents one bucket entry.
     """
 
     def list_func() -> Iterator[File]:
-        config = client_config or {}
-        client, path = Client.parse_url(uri, None, **config)  # type: ignore[arg-type]
+        client, path = Client.parse_url(uri, None, **kwargs)  # type: ignore[arg-type]
         yield from iter_over_async(_scandir(client, path, fetch_workers), get_loop())
 
     return list_func
+
+
+def ls(
+    dc: "DataChain",
+    path: str,
+    recursive: Optional[bool] = True,
+    object_name="file",
+):
+    """
+    Return files by some path from DataChain instance which contains bucket listing.
+    Path can have globs.
+    If recursive is set to False, only first level children will be returned by
+    specified path
+    """
+
+    def _file_c(name: str) -> Column:
+        return Column(f"{object_name}.{name}")
+
+    dc = dc.filter(_file_c("is_latest") == true())
```

Comment:
Q: what is `is_latest`?

Reply:
I think it's related to file versioning in clouds. Latest means it's last version of specific file if cloud versioning is turned on. Honestly I don't think we ever use this, so it's just passed through from the beginning of datachain.

Reply:
was is the similar logic before / in the current implementation?

Reply:
This was basically moved from legacy indexing codebase so nothing has been changed.